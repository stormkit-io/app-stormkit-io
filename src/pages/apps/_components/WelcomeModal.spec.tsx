import type { RenderResult } from "@testing-library/react";
import type { Mock } from "vitest";
import { describe, expect, beforeEach, it, vi } from "vitest";
import { fireEvent, waitFor, render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { WelcomeModal } from "./";

interface Props {
  isOpen: boolean;
  toggleModal: Mock;
}

describe("~/pages/apps/_components/WelcomeModal", () => {
  let wrapper: RenderResult;

  const createWrapper = ({ isOpen, toggleModal }: Props) => {
    wrapper = render(
      <MemoryRouter>
        <WelcomeModal
          welcomeModalId="welcome-modal-id"
          isOpen={isOpen}
          toggleModal={toggleModal}
        />
      </MemoryRouter>
    );
  };

  describe("when is open", () => {
    let toggleModal: Mock;

    beforeEach(() => {
      toggleModal = vi.fn();
      createWrapper({ isOpen: true, toggleModal });
    });

    it("matches the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("closes the modal when close button is clicked", async () => {
      fireEvent.click(wrapper.getByText("Close"));

      await waitFor(() => {
        expect(toggleModal).toHaveBeenCalledWith(false);
      });
    });
  });
});
