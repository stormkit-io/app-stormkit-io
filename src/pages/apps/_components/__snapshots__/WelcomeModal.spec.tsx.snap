// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`~/pages/apps/_components/WelcomeModal > when is open > matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      id="side-bar-root"
    />
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root css-sym2j1-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-4nmryk-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiBox-root css-6f28tg"
        tabindex="-1"
      >
        <div
          class="w-full text-gray-80 max-w-2xl p-4 md:p-8 justify-center h-full md:h-auto flex md:block flex-col md:flex-none MuiBox-root css-12bxit9"
        >
          <h1
            class="font-bold text-xl md:text-3xl text-center mb-8"
          >
            Welcome to Stormkit ðŸŽ‰
          </h1>
          <a
            class="relative flex justify-between items-center shadow w-full p-4 md:p-8 bg-blue-10 hover:bg-black text-gray-80 mb-8 hover:text-white"
            href="https://discord.gg/6yQWhyY"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="text-3xl mr-8"
            >
              <i
                class="fa-brands fa-discord"
                style="color: rgb(88, 101, 242);"
              />
            </span>
            <span
              class="flex flex-col flex-grow"
            >
              <span
                class="font-bold"
              >
                Join our Discord community
              </span>
              <span
                class="text-sm"
              >
                Ask questions and join discussions.
              </span>
            </span>
            <span
              class="fas fa-chevron-right text-base ml"
            />
          </a>
          <a
            class="relative flex justify-between items-center shadow w-full p-4 md:p-8 bg-blue-10 hover:bg-black text-gray-80 hover:text-white"
            href="https://twitter.com/stormkitio"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="text-3xl mr-8"
            >
              <i
                class="fa-brands fa-twitter"
                style="color: rgb(0, 172, 238);"
              />
            </span>
            <span
              class="flex flex-col flex-grow"
            >
              <span
                class="font-bold"
              >
                Follow us on Twitter
              </span>
              <span
                class="text-sm"
              >
                Stay tuned about latest developments on Stormkit.
              </span>
            </span>
            <span
              class="fas fa-chevron-right text-base ml"
            />
          </a>
          <div
            class="mt-8 flex justify-end w-full"
          >
            <button
              class="relative sk-button bg-blue-20 px-6 py-2"
              type="button"
            >
              <span
                class="inline-flex w-full items-center justify-center"
              >
                Close
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
